### DISCUSSION TITLE
Nuxt `<Script>` should support loading strategy

### DISCUSSION THREAD

**Discussion by ms-fadaei** [nuxt/nuxt#18105] [Ideas]
Hi everybody.
As you know Nuxt has a `<Script>` component that is the sugar syntax of `useMeta`. [Next 11 also has `Script` tag.](https://nextjs.org/docs/basic-features/script), but with an important difference: loading strategy for improving performance. They also provide some events for their component. If we can provide a way to do that in Nuxt, it will be very handful 💯. I already do this by plugins in Nuxt 2.
I think this should be implemented by VueUse and we have @antfu here.
I recently create a private repo to implement this with Vue 3 and Vite and planned to implement it in Nuxt (after Nuxt/kit release) but with provided `<Script>` component it is not a good idea anymore.

**Comment by pi0** [comment]
Thanks for bringing up idea @ms-fadaei  👍  Directly adding strategies to [`<Script>` meta tag](https://github.com/nuxt/framework/blob/3e82f0f2cb0c815af898ad78118104a476ee1f8e/packages/nuxt3/src/meta/runtime/components.ts#L149) exposed by nuxt3 app is probably a better idea however as allows us to have better control for integration. Under the hood, we use vueuse/meta as default adoptor

**Reply by ms-fadaei** [reply]
Can I start to do some changes?

**Reply by pi0** [reply]
Of course PR welcome :)
